### PR TITLE
Updated post rows query

### PR DESCRIPTION
**Pull request by BadEnforcer**
Updated the query so instead of querying 2 times, it only does it once.

change are from line 122 to 149

I tried to solve this issue.
https://github.com/LangCorrect/server/issues/303

### CODE REVIEW

File: langcorrect/corrections/views.py [hunk 1]
```diff
@@ -118,11 +120,15 @@ def make_corrections(request, slug):
         is_edit = False
 
         for post_row in all_post_rows:
-            previous_correction = CorrectedRow.available_objects.filter(
-                post_row_id=post_row.id, user=current_user
-            ).first()
+            combined_query = Q(post_row_id=post_row.id, user=current_user)
```
Comment (danielzeljko) [inline langcorrect/corrections/views.py:123]: We're still unfortunately making two queries here, but in a different format (Q objects). What I think @itshesam was hinting at was to select the related objects as we fetch all of the post rows. Django provides a way to do this using [prefetch_related](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#prefetch-related.

The only change we need to make is in **Line 116**:

        all_post_rows = PostRow.available_objects.filter(post=post, is_actual=True).order_by("order")

To:

```
all_post_rows = PostRow.available_objects.filter(
    post=post, is_actual=True
).prefetch_related(
    'correctedrow_set', 
    'perfectrow_set'
).order_by("order")
```

**Comment by BadEnforcer** [comment]
Please check the query and approve if it's viable.

**Comment by danielzeljko** [comment]
Hi @BadEnforcer, sorry for the delay.

Aren't we still doing two queries but in a different way?

I think what @itshesam was thinking about (feel free to correct me if I'm wrong) is to grab all of the post rows then filter the perfect/corrected rows. We would then use these qs and convert them to dicts for more efficient lookups. This is just my assumption. There may be a better approach, so it's open to discussion.

**Comment by itshesam** [comment]
@BadEnforcer 

Your change is not correct, the changed code is going to perform 2 query again not one. To optimize this query, we can fetch all the posts before the for loop using 2 query and then perform the for loop. This reduces our number of query from ```2 * number_of_rows``` to exactly 2.

**Comment by danielzeljko** [comment]
Closing this PR due to no activity within the last few weeks. I'm happy to reopen it if more time is needed.

**Pull request closed without merging**